x/programs/runtime/util_test.go

```diff
@@ -205,36 +210,42 @@ func (t *testRuntime) CallProgram(program codec.Address, function string, params
 		})
 }
 
-func newTestProgram(ctx context.Context, program string) (*testProgram, error) {
+func newTestRuntime(ctx context.Context) *testRuntime {
+	return &testRuntime{
+		Context: ctx,
+		callContext: NewRuntime(
+			NewConfig(),
+			logging.NoLog{}).WithDefaults(CallInfo{Fuel: 10000000}),
+		StateManager: TestStateManager{
+			ProgramsMap: map[string][]byte{},
+			AccountMap:  map[codec.Address]string{},
+			Balances:    map[codec.Address]uint64{},
+			Mu:          test.NewTestDB(),
+		},
+	}
+}
+
+func (t *testRuntime) newTestProgram(program string) (*testProgram, error) {
 	id := ids.GenerateTestID()
 	account := codec.CreateAddress(0, id)
 	stringedID := string(id[:])
 	testProgram := &testProgram{
-		Runtime: &testRuntime{
-			Context: ctx,
-			callContext: NewRuntime(
-				NewConfig(),
-				logging.NoLog{}).WithDefaults(CallInfo{Fuel: 10000000}),
-			StateManager: TestStateManager{
-				ProgramsMap: map[string][]byte{},
-				AccountMap:  map[codec.Address]string{account: stringedID},
-				Balances:    map[codec.Address]uint64{},
-				Mu:          test.NewTestDB(),
-			},
-		},
 		Address: account,
+		Runtime: t,
 	}
-	err := testProgram.Runtime.AddProgram(ProgramID(stringedID), program)
+
+	err := t.AddProgram(ProgramID(stringedID), program)
 	if err != nil {
 		return nil, err
 	}
+	t.AddProgramInstance(ProgramID(stringedID), account)
```

Comment:
I think you can add this logic inside `AddProgram`. I know I said something about functions only doing one thing, but it makes sense to update the programs and accounts maps at the same time.